### PR TITLE
Decision on error handling

### PR DESCRIPTION
**Pull request by zugao**
## Checklist

- [x] Try to isolate changes into separate PRs (to build a better changelog).
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `change`, `decision`, `requirement/quality`, `requirement/functional`, `dependency`
      as they show up in the changelog
- [x] Link this PR to related issues if applicable.

### CODE REVIEW

File: docs/modules/ROOT/pages/explanations/decisions/comp-function-error-handling.adoc [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+* A comprehensive guide on how we should deal with error management.
+* A solution to missing managed resources from desired function when errors occur.
+
+== Proposals
+
+Proposal 1:: Manage Errors in CFs
+
+We don't throw fatal errors but use mostly warnings in our composition functions.
+Using warnings and return right away can remove vital parts of a service.
+In order to mitigate this issue we could define the return error (warning or fatal) directly in the runtime.
```
Comment (Kidswiss) [inline docs/modules/ROOT/pages/explanations/decisions/comp-function-error-handling.adoc:24]: For the sake of the decision: quickly explain the difference between warning and fatal and the implications. Maybe not everyone knows the difference.

Comment (zugao) [reply docs/modules/ROOT/pages/explanations/decisions/comp-function-error-handling.adoc:31]: Added Definition section

File: docs/modules/ROOT/pages/explanations/decisions/comp-function-error-handling.adoc [hunk 2]
```diff
@@ -0,0 +1,71 @@
+= Composition Function Error Handling
+
+In March 2024 we had a severe outage of VSHN PostgreSQL instances.
+This was due to several unfortunate events including our poor handling of errors in composition functions.
+This document will discuss in detail the problem we face with composition functions when errors are not properly handled.
+
+== Problems
+
+* We don't manage errors in our composition functions (CFs) and we assume Crossplane will take care of everything.
```
Comment (Kidswiss) [inline docs/modules/ROOT/pages/explanations/decisions/comp-function-error-handling.adoc:9]: I mean... we do manage errors, but we're not doing a very good job :D

File: docs/modules/ROOT/pages/explanations/decisions/comp-function-error-handling.adoc [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+== Requirements
+
+* A comprehensive guide on how we should deal with error management.
+* A solution to missing managed resources from desired function when errors occur.
+
+== Definitions
+
+* Fatal Result error - the error that stops crossplane from reconciliation which means the desired object will not be applied in the cluster.
+Typically safe move which will ensure no changes to the service but we rarely use this type of error.
```
Comment (Kidswiss) [inline docs/modules/ROOT/pages/explanations/decisions/comp-function-error-handling.adoc:21]: Maybe also mention, that usage of the fatal error at the wrong point will lead to deadlocks. E.g. throwing a fatal error, if the observed object is not yet deployed, which will cause the object to never be deployed at all.

**Review by Kidswiss — APPROVED** [review]

File: docs/modules/ROOT/pages/explanations/decisions/comp-function-error-handling.adoc (hunk 2 above)
Comment (zugao) [reply docs/modules/ROOT/pages/explanations/decisions/comp-function-error-handling.adoc:9]: you are right, added `properly`

**Pull request merged**